wdp/database/connector_wdp.py

```diff
@@ -0,0 +1,36 @@
+import sqlite3
+
+
+class Database:
+    def __init__(self, database_name="wdp_database.db"):
+        # create connection to database
+        self.connection = sqlite3.connect(database_name)
+        self.cursor = self.connection.cursor()
+
+    def __del__(self):
+        # close connection to database after function usage
+        self.connection.close()
+
+    def insert_sql(self, table, *values):
+        # insert values into table
+        self.cursor.execute(f"INSERT INTO {table} VALUES ({','.join(['?' for _ in values])})", values)
```

Comment:
Whenever you have a function that accepts an _iterable_ object (such as a list, a set, tuple, generator...), and you intend to create the _iterable_ using a list comprehension, consider using a generator comprehension.

When you call `function([elem for elem in elements])`,
first thing that is happening when Python is executing this line, is `[elem for elem in elements]`. And this creates a complete list. Then you pass the list to the `function()` that iterates through all the elements inside, which means that the program iterates through provided elements _twice_.
To avoid that (meaning: to optimize that), use generator comprehension, that yields all the values lazily, when the `function()` traverses it: `function(elem for elem in elements)`. Yet so good, it seems even more readable.

If you have any questions on this, feel free to ask them below.